File: examples/numerical_propagation/integrator_method_rk4.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Configuring the RK4 fixed-step integrator method.
"""

import brahe as bh

# RK4: Fixed-step 4th-order Runge-Kutta
config = bh.NumericalPropagationConfig(
    bh.IntegrationMethod.RK4,
    bh.IntegratorConfig.fixed_step(60.0),  # 60 second fixed steps
    bh.VariationalConfig(),
)

print(f"Method: {config.method}")
# Method: IntegrationMethod.RK4

File: examples/numerical_propagation/integrator_method_rk4.rs
//! Configuring the RK4 fixed-step integrator method.

use brahe as bh;

fn main() {
    // RK4: Fixed-step 4th-order Runge-Kutta
    let config = bh::NumericalPropagationConfig {
        method: bh::IntegratorMethod::RK4,
        integrator: bh::IntegratorConfig::fixed_step(60.0), // 60 second fixed steps
        variational: bh::VariationalConfig::default(),
        store_accelerations: true,
        interpolation_method: bh::InterpolationMethod::Linear,
    };

    println!("Method: {:?}", config.method);
    println!("Fixed step: {:?} seconds", config.integrator.fixed_step_size);
    // Method: RK4
    // Fixed step: Some(60.0) seconds
}
